File: generate_pdf.py
"""
Resume PDF Generator
This script converts the resume.html to PDF format
"""

from pathlib import Path

try:
    from weasyprint import HTML, CSS
    
    # Get the resume HTML file
    html_path = Path(__file__).parent / 'resume.html'
    pdf_path = Path(__file__).parent / 'Sonal_Gupta_Resume.pdf'
    
    # Convert HTML to PDF
    HTML(str(html_path)).write_pdf(str(pdf_path))
    print(f"✅ Resume PDF generated successfully: {pdf_path}")
    
except ImportError:
    print("⚠️  WeasyPrint not installed. Installing...")
    import subprocess
    import sys
    subprocess.check_call([sys.executable, "-m", "pip", "install", "weasyprint"])
    
    from weasyprint import HTML, CSS
    
    html_path = Path(__file__).parent / 'resume.html'
    pdf_path = Path(__file__).parent / 'Sonal_Gupta_Resume.pdf'
    
    HTML(str(html_path)).write_pdf(str(pdf_path))
    print(f"✅ Resume PDF generated successfully: {pdf_path}")

except Exception as e:
    print(f"❌ Error: {e}")
    print("\n📝 Alternative: Open resume.html in a browser and use 'Print to PDF'")
    print("   1. Open resume.html in your browser")
    print("   2. Press Ctrl+P (or Cmd+P on Mac)")
    print("   3. Select 'Save as PDF'")
    print("   4. Save as 'Sonal_Gupta_Resume.pdf'")

File: generate_pdf.js
/**
 * Resume PDF Generator using Node.js
 * Install dependencies: npm install puppeteer
 */

const puppeteer = require('puppeteer');
const path = require('path');
const fs = require('fs');

async function generateResumePDF() {
    try {
        const browser = await puppeteer.launch();
        const page = await browser.newPage();
        
        // Load the HTML file
        const htmlPath = `file://${path.resolve(__dirname, 'resume.html')}`;
        
        await page.goto(htmlPath, {
            waitUntil: 'networkidle2'
        });
        
        // Generate PDF
        const pdfPath = path.resolve(__dirname, 'Sonal_Gupta_Resume.pdf');
        
        await page.pdf({
            path: pdfPath,
            format: 'A4',
            margin: {
                top: '0.5in',
                bottom: '0.5in',
                left: '0.5in',
                right: '0.5in'
            },
            printBackground: true
        });
        
        await browser.close();
        
        console.log('✅ Resume PDF generated successfully!');
        console.log(`📄 File saved to: ${pdfPath}`);
        
    } catch (error) {
        console.error('❌ Error generating PDF:', error);
        console.log('\n📝 Alternative: Open resume.html in a browser and use Print to PDF');
        console.log('   1. Open resume.html in your browser');
        console.log('   2. Press Ctrl+P (or Cmd+P on Mac)');
        console.log('   3. Select "Save as PDF"');
        console.log('   4. Save as "Sonal_Gupta_Resume.pdf"');
    }
}

generateResumePDF();
